cloudflare/resource_cloudflare_split_tunnel_exclude.go

```diff
@@ -0,0 +1,127 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceCloudflareSplitTunnelExclude() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourceCloudflareSplitTunnelsExcludeRead,
+		Update: resourceCloudflareSplitTunnelsExcludeUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tunnels": {
+				Computed: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"address": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Description: "The address for the tunnel.",
+						},
+						"host": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Description: "The domain name for the tunnel.",
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Description: "A description for the tunnel.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceCloudflareSplitTunnelsExcludeRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get("account_id").(string)
+
+  splitTunnel, err := client.SplitTunnelExclude(context.Background(), accountID)
+	if err != nil {
+		return fmt.Errorf("Error finding Include Split Tunnels %q", err)
+	}
+
+	tunnelList := make(map[string]interface{}, 0)
+	for _, t := range splitTunnel {
+		tunnelList = append(tunnelList, map[string]interface{}{
+			"address":		  t.Address,
+			"host":			    t.Host,
+			"description":	t.Description,
+		})
+	}
+
+	err = d.Set("tunnels", tunnelList)
+	if err != nil {
+		return fmt.Errorf("error setting Include Split Tunnels: %s", err)
+	}
+	return nil
+}
+
+func resourceCloudflareSplitTunnelsExcludeUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
somewhere (likely towards the end) in this method, we need to call `d.SetID(accountID)` to ensure that Terraform has an identifier in which to compare and validate the changes against. without this identifier, Terraform tests won't ever work as the "anchor" (this ID) is missing from the resource.

Reply:
Reading through the docs, is this something that we need on every method? Even the `read` method in their example has `d.SetID()` at the end here: https://learn.hashicorp.com/tutorials/terraform/provider-setup?in=terraform/providers